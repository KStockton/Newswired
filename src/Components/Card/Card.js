import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const Card = ({ id, author, source, articleImage, description, title}) =>{
  return (
    <section className="CardCard-tile" id={id}>
      <Link to={`/card/${id}`}>
        <img className="Card-img" src={articleImage} alt="travel"/>
      </Link>
      <p className="Card-author">{author}</p>
      <p className="Card-title">{title}</p>
      <p className="Card-description">{description}</p>
    </section>
  );
};

Card.propTypes = {
  author: PropTypes.string,
  source: PropTypes.string,
  title: PropTypes.string,
  articleImage: PropTypes.string,
  description: PropTypes.string,
  content: PropTypes.string
};

export default Card;
