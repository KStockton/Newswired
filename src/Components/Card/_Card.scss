.CardCard-tile {
  border-radius: 8px;
  padding-top: 50px;
  padding-left: 25px;
  font-size: 150%;
  transition: .2s all;
}

.Card-wrapper:hover .CardCard-tile{
  opacity: .5;
  transform: scale(.98);
  box-shadow: none;
}

.Card-wrapper:hover .CardCard-tile:hover{
  filter: blur (0px);
  transform: scale(1.1);
  opacity: 1;
  box-shadow: 0 8px 20px 0 px rgba(0,0,0,0.125);
}

.Card-img {
  height: 162px;
  box-shadow: 2px 4px 6px 2px #a99a9a;
  border-radius: 2px;
  min-width: 304px;
}
.Card-author {
  @include authors( 12px, left, 5px);
  font-family: 'Noto Sans', sans-serif;
  padding-bottom: 16px;
  font-size: 13px;
  font-style: italic;
  color: #ffffff;
}

.Card-title{
  @include titles(.9rem, 1.1875rem, 700, left, 3px);
  padding-bottom: 18px;
  color: #ffffff;
}

.Card-description{
  @include tiles(left, 1rem, 6px );
  font-family: 'Old Standard TT', serif;
  color: #ffffff;
}



