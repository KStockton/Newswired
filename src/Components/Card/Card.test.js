import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';
import { Link } from 'react-router-dom';

describe('Card', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Card/>);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot when props are passed', () => {
    let mockCardImage = "http://cool.com";
    let mockrank = 3;
    let mockauthor = "Michele Obama";
    let mockdescription = 'effor';
    let mocktitle = 'Overcome';
    let mockweeksOnList = 4;


    const wrapper = shallow(<Card
      id={mockrank}
      CardImage={mockCardImage}
      rank={mockrank}
      description={mockdescription}
      author={mockauthor}
      title={mocktitle}
      weeksOnList={mockweeksOnList}
    />);
    
    expect(wrapper).toMatchSnapshot();
  });

  it('Link should have a prop of /card/3 when a user clicks the link', () => {

    let mockId = 3;
    const wrapper = shallow(<Card id={mockId}/>);
    expect(wrapper.find(Link).props().to).toBe('/card/3');
  });
});