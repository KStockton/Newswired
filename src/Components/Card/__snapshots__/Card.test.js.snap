// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
<section
  className="CardCard-tile"
>
  <img
    alt="travel"
    className="Card-img"
  />
  <p
    className="Card-author"
  />
  <p
    className="Card-title"
  />
  <p
    className="Card-description"
  />
</section>
`;

exports[`Card should match the snapshot when props are passed 1`] = `
<section
  className="CardCard-tile"
>
  <img
    alt="travel"
    className="Card-img"
  />
  <p
    className="Card-author"
  >
    Michele Obama
  </p>
  <p
    className="Card-title"
  >
    Overcome
  </p>
  <p
    className="Card-description"
  >
    effor
  </p>
</section>
`;
